docs/devel/offline-virtual-machine.md

```diff
@@ -0,0 +1,291 @@
+documentation.
+
+## OfflineVirtualMachine behaviour
+
+The OfflineVirtualMachine has to be in sync with its VirtualMachine. This means
+that the OfflineVirtualMachine controller has to observe both, the OfflineVirtualMachine
+and created VirtualMachine. When the [link](#ownerreference) is established the config changes
+are translated to the VirtualMachine, and coresponding status changes are
+translated back to OfflineVirtualMachine.
+
+TBD this needs to be more specific
+
+### Status
+
+The status of the VirtualMachine is watched and is reflected in OfflineVirtualMachine status.
+The status contains the current state of the VirtualMachine (running/not running)
+and whether it is ready.
+
+More detailed information is queried directly from the status of the VirtualMachine.
+
+### OwnerReference
+
+Linking the created VirtualMachine to its parent OfflineVirtualMachine pose
+a challenge. Using the same name is only part of the solution. To find the
+parent OfflineVirtualMachine programatically in the Kubernetes, the OwnerReference
+is used. As described in the
+[design document](https://github.com/kubernetes/community/blob/master/contributors/design-proposals/api-machinery/controller-ref.md),
+the OwnerReference lives in the metadata section of the object and is created
+automaticaly. Example:
+
+```YAML
+apiVersion: kubevirt.io/v1alpha1
+kind: VirtualMachine
+metadata:
+  name: myvm
+  ownerReferences:
+    - controller: true
+      uid: 1234-1234-1234-1234
+      kind: OfflineVirtualMachine
+      version: kubevirt.io/v1alpha1
+```
+
+### Update strategy
+
+For now implicit: OnDelete. Can later be extended to RollingUpdate if needed.
+Spec changes have no direct effect on already running VMs, and they will not
+directly be propagated to the VM. If a VM should be running (spec.running=true)
+and it is powered down (VM object delete, OS shutdown, ...),
+the VM will be re-created by the controller with the new spec.
+
+### Delete strategy
+
+The delete has a cascade that deletes the created VirtualMachine. If a cascade
+is turned off the VirtualMachine is orphaned and leaved running.
+When the OfflineVirtualMachine with the same name as orphaned VirtualMachine
+is created, the VirtualMachine gets adopted and OwnerReference
+is updated accordingly.
+
+### Reset and Reboot
+
+This is not covered by the OfflineVirtualMachine. This functionality shall
+be achieved by subresources for the VirtualMachine (imperative),
+and will not result in a recreation of the VirtualMachine object or its Pod.
+From the KubeVirt perspective, the VirtualMachine is running all the time.
+Thus spec changes on OfflineVirtualMachine will not be propagated to
+the VirtualMachine in that case.
+
+## How it is implemented
+
+The first implementation of the OfflineVirtualMachine kind has two parts:
+
+1) The OfflineVirtualMachine custom resource definition,
+2) The controller watching for the changes and updating the state.
```

Comment:
Where will this controller live? Separate pod, or internal to the virt-controller?